packages/unified-latex-to-pretext/libs/pretext-subs/to-pretext.ts

```diff
@@ -59,48 +64,98 @@ export function toPretextWithLoggerFactory(
                 return x("m", printRaw(node.content));
             case "mathenv":
             case "displaymath":
-                return x("m", printRaw(node.content));
+                return x("me", printRaw(node.content));
             case "verb":
             case "verbatim":
-                return x("pre", { className: node.env }, node.content);
+                return x("pre", node.content);
             case "whitespace":
                 return { type: "text", value: " ", position: node.position };
             case "parbreak":
-                return x("br");
+                // warn first
+                logger(
+                    `There is no equivalent for parbreak, so it was replaced with an empty string.`,
+                    node
+                );
+
+                // return an empty string
+                return {
+                    type: "text",
+                    value: "",
+                    position: node.position,
+                };
             case "group":
                 // Groups are just ignored.
                 return node.content.flatMap(toPretext);
             case "environment":
+                // check if it's a new environment made to replace a division node
+                if (isMappedEnviron(node)) {
+                    // get the division macro associated with this node
+                    let divisionName = divisions.find(
+                        (x) => x.mappedEnviron === node.env
+                    )?.division;
+
+                    // for subparagraph, give a warning since pretext has no equivalent tag
+                    if (divisionName === "subparagraph") {
+                        logger(
+                            `Warning: There is no equivalent tag for "subparagraph", "paragraphs" was used as a replacement.`,
+                            node
+                        );
+                    }
+
+                    // paragraph and subparagraph become paragraphs
+                    if (
+                        divisionName === "paragraph" ||
+                        divisionName === "subparagraph"
+                    ) {
+                        divisionName = "paragraphs";
+                    }
+
+                    // create a title tag containing the division macro's title arg
+                    const title = getArgsContent(node)[0];
+
+                    const titleTag = x("title", title?.flatMap(toPretext));
+
+                    if (divisionName && title) {
```

Comment:
Should always happen if we have a division name. if `title` is `null`, we should log a warning. Then inserting the empty `titleTag` will be fine.